relay-server/src/utils/garbage.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::mpsc;
+use std::thread::JoinHandle;
+
+pub struct GarbageDisposal<T> {
+    tx: Option<mpsc::Sender<T>>,
+    join_handle: Option<JoinHandle<()>>,
+}
+
+impl<T: Send + 'static> GarbageDisposal<T> {
+    pub fn new() -> Self {
+        let (tx, rx) = mpsc::channel();
+
+        let join_handle = std::thread::spawn(move || {
+            relay_log::debug!("Start garbage collection thread");
+            while let Ok(object) = rx.recv() {
+                // TODO: Log size of channel queue as a gauge here
```

Comment:
afaik the only way to do this is by having your own `AtomicU32` to keep track of the number of entries inside it.  The other metrics option is to count on both ends, then you'll have to rates which you can subtract and no need to handle an atomic.

Reply:
question is if you need this at all.  the risk you are trying to mitigate is a disposal thread that doesn't keep up and the channel growing indefinitely.  Slowing down the channel is not helpful with that, and neither is having a metric: it will only allow you to figure out eventually that there's a metric growing together with the memory.

The way to really mitigate is to make a bounded channel: if the channel is full you can log the error to sentry and you start dropping in the sender thread again.

Reply:
Sorry I somehow missed this comment and added the metric anyway. The bounded channel idea makes sense to me, let's see what values the metric shows and then follow up if necessary.